api/index.js

```diff
@@ -1,7 +1,22 @@
 'use strict'
 var path = require('path');
+var fs = require('fs');
 var args = process.argv;
 if (!(args.length == 4 && args[2] == "-c" && args[3] != "")) {
-    throw new Error("missing config file\nUsage:use '-c' option to specify the config file path");
+  throw new Error("missing config file\nUsage:use '-c' option to specify the config file path");
 }
-var apiServer = require(path.join(__dirname, 'app.js'))(args[3]);
+var configFilePath = args[3];
+
+if (!configFilePath || !fs.existsSync(configFilePath)) {
+      logger.error("Invalid configuration file path: " + configFilePath);
+      throw new Error('configuration file does not exist:' + configFilePath);
+  }
+var settings = require(path.join(__dirname, './lib/config/setting.js'))((JSON.parse(
+        fs.readFileSync(configFilePath, 'utf8'))));
+var dbErrorCallback = function(err) {
+  if (err) {
+    logger.error('DB configuration is incorrect, server will exit', err);
+    process.exit(1);
```

Comment:
when setting is invalid, api server exit with " throw new Error("Invalid public TLS ca certificate path: " + settings.publicTls.caCertFile);" instead of "process.exit(1)
should we handle these kinds of  FATAL error with a consistent approach?